pkg/cmd/pipeline/translate.go

```diff
@@ -189,6 +195,30 @@ func UpdateEnvs(ctx context.Context, name, namespace string, envs []string, c ku
 	return nil
 }
 
+func AddPhase(ctx context.Context, name, namespace, phase string, duration time.Duration, c kubernetes.Interface) error {
+	cmap, err := configmaps.Get(ctx, TranslatePipelineName(name), namespace, c)
+	if err != nil {
+		return err
+	}
+	val, ok := cmap.Data[phasesField]
+	phases := []phaseJSON{}
+	if ok {
+		if err := json.Unmarshal([]byte(val), &phases); err != nil {
+			return err
+		}
+	}
+	phases = append(phases, phaseJSON{
+		Name:     phase,
+		Duration: duration.Seconds(),
+	})
+	encodedPhases, err := json.Marshal(phases)
+	if err != nil {
+		return err
+	}
+	cmap.Data[phasesField] = string(encodedPhases)
```

Comment:
Would it make sense to encode the data to base64 as we do with other data like variables?

Reply:
I don't think it's necessary given that this data is not going to grow too much in the future and from the k8s spec it should be human readable.

Reply:
It is not about the length, it is about the format and possible issues with chars or similar. As we are controlling the format of that, there might not be issues, but just in case. Anyways, this is completely a nit, you can ignore it

Reply:
Right now it's a json that we generate. We'll have to rethink it if we add the commands names but right now we are the ones that set all the naming in the json generated